assembly/src/todo/instruction/flow_ops.rs

```diff
@@ -0,0 +1,23 @@
+use vm_core::{code_blocks::CodeBlock, Operation};
+
+use crate::{
+    todo::{span_to_block, AssemblerContext},
+    AssemblerError,
+};
+
+// FLOW CONTROL OPERATIONS
+// ================================================================================================
+
+pub(super) fn exec_local(
+    context: &mut AssemblerContext,
+    blocks: &mut Vec<CodeBlock>,
+    span: &mut Vec<Operation>,
+    idx: u16,
+) -> Result<(), AssemblerError> {
+    span_to_block(blocks, span);
+    context
+        .locals
+        .get(idx as usize)
+        .map(|p| blocks.push(p.code_root().clone()))
+        .ok_or_else(|| AssemblerError::undefined_proc(idx))
+}
```

Comment:
As mentioned in one of prior comments, I'd change this to be something like this:
```Rust
pub(super) fn exec_local(
    proc_idx: u16,
    context: &AssemblerContext,
) -> Result<Option<CodeBlock>, AssemblerError> {
    context
        .locals
        .get(proc_idx as usize)
        .map(|p| Some(p.code_root().clone()))
        .ok_or_else(|| AssemblerError::undefined_proc(idx))
}
```
Then, the closing of the span block could be handled in the `compile_body()` function.

Reply:
Refactored